design/authn_azure/authn_azure_solution_design.md

```diff
@@ -0,0 +1,373 @@
+  authn-azure/user-assigned-identity: test-app-pipeline 
+OR 
+  authn-azure/system-assigned-identity: ie_qWCXhXxt1zIEsu4c7acQVGn4
+```
+
+An example of a "xms_mirid" claim is listed below. We will use what is defined in this claim and compare it against the Conjur identity for the Azure resource. If they match, then we can confirm that the 
+Azure resource that has made the request has a Conjur identity.
+
+```yaml
+"xms_mirid": "/subscriptions/<subscription_id>/resourcegroups/<resource_group>/providers/Microsoft.ContainerInstance/containerGroups/<assigned_identity>"
+```
+
+| Host annotation                                                      | xms_mirid              |
+|----------------------------------------------------------------------|------------------------|
+| `authn-azure/subscription-id`                                        | subscription_id        |
+| `authn-azure/resource-group`                                         | resource_group         |
+| `authn-azure/user-assigned-identity` or `system-assigned-identity`   | assigned_identity      |
+
+### Backwards compatibility
+
+None because this is a new feature.
+
+### Performance
+
+Azure authenticator performance should conform with our other authenticators with an average call time of 1 second.
+
+### Affected Components
+
+- Conjur
+
+- DAP
+
+## Test Plan
+
+TBT
+
+### Integration tests
+
+#### Functionality
+
+This suite of tests assumes that the Azure authenticator has been configured correctly
+
+|    | Given                                                                                        | When                                                     | Then                                                                                          | Manual / UT / Integration | Status     |
+|----|----------------------------------------------------------------------------------------------|----------------------------------------------------------|-----------------------------------------------------------------------------------------------|---------------------------|------------|
+| 1  | *Vanilla flow* - a Conjur identity exists for Azure VM and added to privileged layer         | VM sends authentication request with a valid Azure token | Conjur access token is returned in body and action is logged                                  | Integration               |            |
+| 2  | A Conjur Host identity for Azure VM, added to privileged layer and has permissions on secret | VM sends request with valid Azure token to fetch secret  | Secret is successfully returned and action is logged                                          | Integration               |            |
+| 3  | No Conjur Host identity exists for Azure VM                                                  | VM sends authentication request with a valid Azure token | 403 Forbidden response / RoleNotFound in error body is returned and action is logged          | Integration               |            |
```

Comment:
1. Why would we return a 403 in this case? In case the user does not exist we return a 401. 403 is in case the user exists but is not permitted on the webservice.
2. We do not return the error in the response body, only write it to the log.